packages/directlinespeech/src/createAdapters.js

```diff
@@ -0,0 +1,161 @@
+    console.warn(
+      'botframework-directlinespeech: Custom Speech is currently not supported, ignoring speechRecognitionEndpointId.'
+    );
+  }
+
+  if (speechSynthesisDeploymentId) {
+    console.warn(
+      'botframework-directlinespeech: Custom Voice is currently not supported, ignoring speechSynthesisDeploymentId.'
+    );
+  }
+
+  if (speechSynthesisOutputFormat) {
+    console.warn(
+      'botframework-directlinespeech: Synthesis output format is currently not supported, ignoring speechSynthesisOutputFormat.'
+    );
+  }
+
+  if (textNormalization) {
+    console.warn(
+      'botframework-directlinespeech: Text normalization is currently not supported, ignoreing "textNormalization".'
+    );
+  }
+
+  if (userID || username) {
+    console.warn(
+      'botframework-directlinespeech: Custom "userId" and "username" are currently not supported, ignoring.'
+    );
+  }
+
+  let config;
+
+  if (authorizationToken) {
+    config = BotFrameworkConfig.fromAuthorizationToken(authorizationToken, region);
+  } else {
+    config = BotFrameworkConfig.fromSubscription(subscriptionKey, region);
+  }
+
+  // Supported options can be found in DialogConnectorFactory.js.
+
+  config.setProperty(PropertyId.SpeechServiceConnection_RecoLanguage, speechRecognitionLanguage);
+
+  // None of the following works for setting output format.
+
+  // config.setProperty(PropertyId.SpeechServiceResponse_OutputFormatOption, OutputFormat[OutputFormat.Detailed]);
+  // config.setProperty(PropertyId.SpeechServiceResponse_RequestDetailedResultTrueFalse, true);
+  // config.setProperty(OutputFormatPropertyName, OutputFormat[OutputFormat.Detailed]);
+  // config.setServiceProperty(PropertyId.SpeechServiceResponse_RequestDetailedResultTrueFalse, "true", ServicePropertyChannel.UriQueryParameter);
+
+  // The following code is copied from C#, it should set from.id, but it did not.
+  // https://github.com/Azure-Samples/Cognitive-Services-Direct-Line-Speech-Client/blob/master/DLSpeechClient/MainWindow.xaml.cs#L236
+  userID && config.setProperty(PropertyId.Conversation_From_Id, userID);
+
+  // The following code is copied from C#, and it is not working yet.
+  // https://github.com/Azure-Samples/Cognitive-Services-Direct-Line-Speech-Client/blob/master/DLSpeechClient/MainWindow.xaml.cs
+  // speechRecognitionEndpointId && config.setServiceProperty('cid', speechRecognitionEndpointId, ServicePropertyChannel.UriQueryParameter);
+  // speechSynthesisDeploymentId && config.setProperty(PropertyId.conversation_Custom_Voice_Deployment_Ids, speechSynthesisDeploymentId);
+
+  const dialogServiceConnector = patchDialogServiceConnectorInline(new DialogServiceConnector(config, audioConfig));
+
+  dialogServiceConnector.connect();
+
+  // Renew token
+  if (authorizationToken) {
+    const interval = setInterval(async () => {
+      // If the connector has been disposed, we should stop renewing the token.
+      // TODO: We should use a public implementation if Speech SDK has one.
```

Comment:
file issue? put Tracking: in the title. Maybe also add a Tracking label for us to check periodically. Or, we could use the Blocked label.

Reply:
Also, add the External: Cognitive Services label 😎

Reply:
Done. https://github.com/microsoft/BotFramework-WebChat/issues/2660